tests/queries/0_stateless/01882_total_rows_approx.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+$CLICKHOUSE_CLIENT -q "drop table if exists data_01882"
+$CLICKHOUSE_CLIENT -q "create table data_01882 (key Int) Engine=MergeTree() partition by key order by key as select * from numbers(10)"
+QUERY_ID="$CLICKHOUSE_TEST_NAME-$(tr -cd '[:lower:]' < /dev/urandom | head -c10)"
+
+function check_background_query()
+{
+    echo "Waiting for query to be started..."
+    while [[ $($CLICKHOUSE_CLIENT --param_query_id="$QUERY_ID" -q 'select count() from system.processes where query_id = {query_id:String}') != 1 ]]; do
+        sleep 0.01
+    done
+    echo "Query started."
+
+    echo "Checking total_rows_approx."
+    # check total_rows_approx multiple times
+    # (to make test more reliable to what it covers)
+    local i=0
+    for ((i = 0; i < 20; ++i)); do
+        $CLICKHOUSE_CLIENT --param_query_id="$QUERY_ID" -q 'select total_rows_approx from system.processes where query_id = {query_id:String}'
```

Comment:
I don't think it is a reliable test.
The point is that we add processlist entry first, and only then build pipeline and add `total_rows_approx`.
Even if we update processlist fast, it is not guaranteed that query has initialized for any timeout.

Also, I don't understand why we check it 20 times, if result is expected to be 10 for each call. `total_rows_approx`  cannot decrease

Reply:
The sleepEachRow will be executed 13 times:
- 10 for 10 rows
- 1 for primary key analysis
- 2 for partition pruning

That said that the query will appears in the process list before all parts will be accounted in total_rows_approx (so total_rows_approx will be 0).
So that 20 iterations is just to ensure that we will get last 10 values, and in all of them total_rows_approx should be 10, one last value is not enough since there total_rows_approx=10 even before this patch, and it is 10 just to ensure that for all blocks we have the same total_rows_approx=10 in process list.
It is 20 since server may randomly stuck (by some fsync from some thread or similar), and 20 seconds looks good enough.

Reply:
Ok, now I see that we use `tail -n 5` for it. It is better then (however, I still don't like we have sleeps).

So, as I understand, previously (for single thread) we could get results like `0 0 0 1 2 3 4 5 6 7 8 9 10 10`, and now we should have something like `0 0 0 10 10 10 10 10 10 10 10 10 10 10`, and we want to distinguish it.
Looks like we can also exit the loop if query returned empty result.

Could we also decrease timeout up to e.g. 0.1 or even less (in both sleep and select)?

Reply:
Also, could you add a comment about what do we want to test?

Reply:
>So, as I understand, previously (for single thread) we could get results like 0 0 0 1 2 3 4 5 6 7 8 9 10 10, and now we should have something like 0 0 0 10 10 10 10 10 10 10 10 10 10 10, and we want to distinguish it.

Correct.

>Could we also decrease timeout up to e.g. 0.1 or even less (in both sleep and select)?

I used 1 seconds, since 0.1 is not very determine, but I'm fine with 0.1

Reply:
>Looks like we can also exit the loop if query returned empty result.

Yep, but this does not looks major and will only make the test a little bit more complex.

> Could we also decrease timeout up to e.g. 0.1 or even less (in both sleep and select)?

0.1 is not enough, since in this case the query may exit before it will be checked in background (in release build it will be too fast, and in debug build the client will be too slow).

Reply:
>Also, could you add a comment about what do we want to test?

Done.